### ISSUE TITLE
[excludeDays] doesn't work correctly in the week view component

### ISSUE THREAD

**Issue by ezodagrom** [mattlewis92/angular-calendar#973]
**Describe the bug**

When using some combinations of days to be excluded, some days that shouldn't be excluded are excluded anyway, for example, setting [excludeDays]="[4, 6]", friday (day 5) is also excluded in the week view (even though it works fine in the month view).

Seems like the combinations of days "0, 2" and "4, 6" are the problematic ones for me, but I'm not sure if there's others combinations with issues as well.

Including a screenshot using [excludeDays]="[4, 6]", in both month and week views (friday is missing in the week view). The screenshot was taken in a new project:

**Minimal reproduction of the problem with instructions**

**Screenshots**

![angular calendar bug report](https://user-images.githubusercontent.com/5478085/57298176-7975a000-70c9-11e9-8418-22e4c0ecf06d.png)

**Versions**

- `@angular/core`: 7.2.0
- `angular-calendar`: 0.27.7
- Browser name and version:
Firefox 66.0.4, Google Chrome 74.0.3729.131, Microsoft Edge 44.17763.1.0

**Comment by evilstiefel** [comment]
On a related note, if you specify excludedDays e.g. to be `[0, 1, 2, 3, 4, 5]`, so only one day (to "simulate" a day view), the component also crashes. Here, the error comes from `getWeekViewPeriod()`, where **excluded** seems to be null. Can open a separate issue for that, as well.

**Comment by mattlewis92** [comment]
@evilstiefel it's probably related to this bug, I'll make sure to include a test case for that when I get round to fixing it. As a workaround you can set `[daysInWeek]="1"` to get the same effect.

**Comment by evilstiefel** [comment]
From a quick check, it seems that the **calendar-utils** function for getWeekView() is initialized with wrong values for these instances.

`viewStart` is assigned a value *after* `viewDate` and so the function `getWeekViewHeader()` returns an empty array. If I manually check and assign the earlier date, it works well, e.g.:

```ts
viewStart = (viewStart <= viewDate) ? viewStart : viewDate;
viewStart = startOfDay(viewStart)
```
The calls from angular-calendar to that function don't even seem to include values for viewStart and viewEnd, so the initializer-values are assigned (as far as I can tell, only glanced over it briefly).

**Comment by mattlewis92** [comment]
Finally managed to fix this one, please give `0.27.17` a go 😄
